### ISSUE TITLE
DependenciesTreeViewProvider Exception When Opening Solutions

### ISSUE THREAD

**Issue by BenVillalobos** [dotnet/project-system#5454]
**Visual Studio Version**:
16.4.0 Preview 1.0 [29306.174.master]

**Summary**:
Visual studio runs into "unexpected problems" when booting up and opening a solution that previously launched just fine. 

![image](https://user-images.githubusercontent.com/4691428/64572667-2af93c00-d31d-11e9-817f-72f1ffe16c07.png)

The log files show an exception seemingly caused at `DependenciesProjectTreeViewProvider`
```
Microsoft.Assumes+InternalErrorException: An internal error occurred. Please contact Microsoft Support.
   at Microsoft.Assumes.Fail(String message)
   at Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.DependenciesTreeViewProvider.CreateOrUpdateNode(IProjectTree node, IDependencyViewModel viewModel, IRule browseObjectProperties, Boolean isProjectItem, Nullable`1 additionalFlags, Nullable`1 excludedFlags)
   at Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.DependenciesTreeViewProvider.<>c__DisplayClass6_0.<<BuildTreeAsync>g__BuildMultiTargetTreeAsync|2>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
```

**Steps to Reproduce**:

Update to latest internal VS (version shown above) and open a solution (in my case this was MSBuild). Immediately after the project opens the errors shown in the picture above will appear. This issue didn't happen in a fresh console or windows project. 

See the attached log for more details
[VsProjectFault_41d05fe1-d892-4e2c-bd60-bc96ee270984.failure.txt](https://github.com/dotnet/project-system/files/3592963/VsProjectFault_41d05fe1-d892-4e2c-bd60-bc96ee270984.failure.txt)

**Comment by drewnoakes** [comment]
This bug was introduced in 8f558fe5d4f6891164601017b48d15a7bdb83ea6. `filePath` can be `null` for non-project-items. PR inbound.

**Comment by drewnoakes** [comment]
Fixed in #5455.
